lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,336 @@
+
+  delete testPdr.execution;
+  delete testPdr.PANSent;
+  delete testPdr.PANmessage;
+  delete testPdr.progress;
+  delete testPdr.stats;
+  delete testPdr.address;
+  delete testPdr.originalUrl;
+  delete testPdr.timestamp;
+  delete testPdr.duration;
+  delete testPdr.updatedAt;
+
+  await migratePdrRecord(testPdr, knex);
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      ...testPdr,
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
+      status: testPdr.status,
+      progress: null,
+      pan_sent: null,
+      pan_message: null,
+      stats: null,
+      address: null,
+      original_url: null,
+      timestamp: null,
+      duration: null,
+      created_at: new Date(testPdr.createdAt),
+      updated_at: new Date(testPdr.updatedAt),
+    }, ['createdAt', 'updatedAt', 'pdrName', 'collection', 'provider', 'collectionId', 'updated_at'])
+  );
+});
+
+test.serial('migratePdrRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+  await migratePdrRecord(testPdr, knex);
+
+  await t.throwsAsync(
+    migratePdrRecord(testPdr, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migratePdrs skips already migrated record', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+
+  await migratePdrRecord(testPdr, knex);
+  await pdrsModel.create(testPdr);
+  const migrationSummary = await migratePdrs(process.env, knex);
+  t.deepEqual(migrationSummary,
+    {
+      dynamoRecords: 1,
+      skipped: 1,
+      failed: 0,
+      success: 0,
+    });
+  const records = await t.context.knex.queryBuilder().select().table(tableNames.pdrs);
+  t.is(records.length, 1);
+  t.teardown(() => pdrsModel.delete({ pdrName: testPdr.pdrName }));
+});
+
+test.serial('migratePdrs processes multiple PDR records', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+  const anotherCollection = fakeCollectionFactory();
+  const anotherProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+  });
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+  const anotherPdr = generateTestPdr(anotherCollection.name, anotherProvider.id);
+  await migrateFakeCollectionRecord(anotherCollection, t.context.knex);
+  await migrateFakeProviderRecord(anotherProvider, t.context.knex);
```

Comment:
I'd prefer to not be implicitly testing the collection/provider migration code as part of the PDR migration code tests.

Instead, let's use `fakeCollectionRecordFactory` to set `t.context.testCollection` and rely on `test.beforeEach` to manually create those objects using the model methods. And do the same thing for provider